HW05-byteCodes/src/main/java/ru/mts/MyIOC.java

```diff
@@ -0,0 +1,62 @@
+package ru.mts;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MyIOC {
+    static ArrayList<Method> loggedMethods;
+
+    public MyIOC() {
+    }
+
+    private static List<Method> getAnnotationMethods(Object object, Class classAnnotation) {
+        ArrayList<Method> varAnnotationMethod = new ArrayList(10);
+        for (Method method : object.getClass().getDeclaredMethods()) {
+            if (method.isAnnotationPresent(classAnnotation)) {
+                varAnnotationMethod.add(method);
+            }
+        }
+        return (List<Method>) varAnnotationMethod;
```

Comment:
```suggestion
        List<Method> annotatedMethods = new ArrayList(10);
        for (Method method : object.getClass().getDeclaredMethods()) {
            if (method.isAnnotationPresent(classAnnotation)) {
                annotatedMethods.add(method);
            }
        }
        return annotatedMethods;
```

Используйте везде интерфейс.
"var" в названии переменной лишнее (в другом месте еще сами посмотрите этот момент).